uniqueness/leaderelection.go

```diff
@@ -48,10 +46,9 @@ type LeaderelectionManager struct {
 	kubeClient k8s.Interface
 }
 
-func NewLeaderElector(log *logp.Logger, cfg *config.Config, k8sProvider providers.KubernetesClientGetter) Manager {
-	kubeClient, err := k8sProvider.GetClient(log, cfg.KubeConfig, kubernetes.KubeClientOptions{})
-	if err != nil {
-		log.Errorf("NewLeaderElector error in GetClient: %v", err)
+func NewLeaderElector(log *logp.Logger, client k8s.Interface) Manager {
+	if client == nil {
+		log.Errorf("NewLeaderElector no k8s client received")
```

Comment:
This will lead to another error log even though it's expected after `GetK8sClient()` fails. I would remove the log to avoid confusion.

```suggestion
```

Reply:
I changed the log msg

Reply:
I believe that since we already logged the error previously, this error log is not really neede anymore and it's confusing